File: app/src/main/java/net/mieczkowski/yelpbusinessexample/recyclerAdapters/searchBusiness/SearchBusinessViewHolder.java
package net.mieczkowski.yelpbusinessexample.recyclerAdapters.searchBusiness;

import android.graphics.Color;
import android.graphics.drawable.Drawable;
import android.support.v4.content.ContextCompat;
import android.support.v4.graphics.drawable.DrawableCompat;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.squareup.picasso.Picasso;

import net.mieczkowski.yelpbusinessexample.R;
import net.mieczkowski.yelpbusinessexample.models.business.YelpBusiness;
import net.mieczkowski.yelpbusinessexample.recyclerAdapters.base.BaseViewHolder;

import butterknife.BindView;

/**
 * Created by Josh Mieczkowski on 10/22/2017.
 */

public class SearchBusinessViewHolder extends BaseViewHolder<YelpBusiness>{

    @BindView(R.id.imgIcon)
    ImageView imgIcon;

    @BindView(R.id.txtTitle)
    TextView txtTitle;

    @BindView(R.id.txtRatings)
    TextView txtRatings;

    @BindView(R.id.txtReviewCount)
    TextView txtReviewCount;


    public SearchBusinessViewHolder(View itemView) {
        super(itemView);

        Drawable ratingsDrawable = ContextCompat.getDrawable(itemView.getContext(), R.drawable.ic_star_rate_white_18dp);
        ratingsDrawable = DrawableCompat.wrap(ratingsDrawable);
        DrawableCompat.setTint(ratingsDrawable, Color.parseColor("#FFD700"));
        txtRatings.setCompoundDrawablesWithIntrinsicBounds(ratingsDrawable, null, null, null);

        Drawable reviewCountDrawable = ContextCompat.getDrawable(itemView.getContext(), R.drawable.ic_edit_black_18dp);
        reviewCountDrawable = DrawableCompat.wrap(reviewCountDrawable);
        DrawableCompat.setTint(reviewCountDrawable, Color.parseColor("#A9A9A9"));
        txtReviewCount.setCompoundDrawablesWithIntrinsicBounds(reviewCountDrawable, null, null, null);
    }

    @Override
    public void onBind(YelpBusiness object) {

        if(object.getBusinessDetails().getImgUrl() != null && !object.getBusinessDetails().getImgUrl().isEmpty()) {
            Picasso.with(itemView.getContext())
                    .load(object.getBusinessDetails().getImgUrl())
                    .into(imgIcon);
        }else{
            imgIcon.setImageResource(R.drawable.ic_close_black_24dp);
        }

        txtTitle.setText(object.getName());

        txtRatings.setText(String.valueOf(object.getBusinessDetails().getRating()));
        txtReviewCount.setText(String.valueOf(object.getBusinessDetails().getReviewCount()));
    }
}

File: app/src/main/java/net/mieczkowski/yelpbusinessexample/recyclerAdapters/searchBusiness/SearchBusinessViewHolder.kt
package net.mieczkowski.yelpbusinessexample.recyclerAdapters.searchBusiness

import android.graphics.Color
import android.support.v4.content.ContextCompat
import android.support.v4.graphics.drawable.DrawableCompat
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import kotlinx.android.synthetic.main.row_seach_business.view.*
import net.mieczkowski.dal.services.businessLookupService.models.YelpBusiness
import net.mieczkowski.yelpbusinessexample.R
import net.mieczkowski.yelpbusinessexample.exts.loadUrl
import net.mieczkowski.yelpbusinessexample.recyclerAdapters.base.BaseViewHolder

/**
 * Created by Josh Mieczkowski on 10/22/2017.
 */

class SearchBusinessViewHolder(itemView: View) : BaseViewHolder<YelpBusiness>(itemView) {

    val imgIcon: ImageView = itemView.imgIcon

    val txtTitle: TextView = itemView.txtTitle
    val txtRatings: TextView = itemView.txtRatings
    val txtReviewCount: TextView = itemView.txtReviewCount

    init {
        ContextCompat.getDrawable(itemView.context, R.drawable.ic_star_rate_white_18dp)?.let {
            val ratingsDrawable = DrawableCompat.wrap(it)

            DrawableCompat.setTint(ratingsDrawable, Color.parseColor("#FFD700"))
            txtRatings.setCompoundDrawablesWithIntrinsicBounds(ratingsDrawable, null, null, null)
        }

        ContextCompat.getDrawable(itemView.context, R.drawable.ic_edit_black_18dp)?.let {
            val reviewCountDrawable = DrawableCompat.wrap(it)

            DrawableCompat.setTint(reviewCountDrawable, Color.parseColor("#A9A9A9"))
            txtReviewCount.setCompoundDrawablesWithIntrinsicBounds(reviewCountDrawable, null, null, null)
        }
    }

    override fun onBind(item: YelpBusiness?) {
        item?.let {
            imgIcon.loadUrl(it.businessDetails?.imgUrl)

            txtTitle.text = it.name

            txtRatings.text = it.businessDetails?.rating.toString()
            txtReviewCount.text = it.businessDetails?.reviewCount.toString()
        }

    }
}
